jbpm-wb-common/jbpm-wb-common-client/src/main/java/org/jbpm/workbench/common/client/list/AbstractMultiGridView.java

```diff
@@ -404,4 +423,23 @@ public void onResize() {
             listTable.onResize();
         }
     }
-}
\ No newline at end of file
+
+    @Override
+    public ColumnSortList reloadColumnSortList() {
+        ColumnSortList columnSortList = getListGrid().getColumnSortList();
+        GridPreferencesStore gridPreferencesStore = listTable.getGridPreferencesStore();
+        GridSortedColumnPreference gridSortedColumnPreference = gridPreferencesStore.getGridSortedColumnPreference();
+        if (gridPreferencesStore != null && gridSortedColumnPreference != null && columnSortList.size() <= 1) {
+            Optional<ColumnMeta<T>> optional = getListGrid().getColumnMetaList().stream().filter(
+                    tColumnMeta -> tColumnMeta.getColumn().getDataStoreName().equals(gridSortedColumnPreference.getDataStoreName()))
+                    .findFirst();
+            if (optional.isPresent()) {
+                Column col = optional.get().getColumn();
+                col.setDataStoreName(gridSortedColumnPreference.getDataStoreName());
+                columnSortList.push(new ColumnSortList.ColumnSortInfo(col,
```

Comment:
perhaps you should do _columnSortList.clear_ before pushing the new column to ensure it doesnt get appended.

Reply:
I'm not sure either, it might cause some issue, but it seems to me that in this case you are adding another sorting column, when we should ensure, that only one column is sorted at time.

Reply:
In here https://github.com/kiegroup/jbpm-wb/blob/master/jbpm-wb-common/jbpm-wb-common-client/src/main/java/org/jbpm/workbench/common/client/list/AbstractMultiGridPresenter.java#L191
, just have one column is sorted at time.

Reply:
columnSortList can be more than 1, it have two cases can call columnSortList.push, one click a column head and reload GridSortedColumnPreference from GridPreferencesStore when init the table. But if user just click column head , the action will also reload GridSortedColumnPreference action and will throw exception, so I ignored columnSortList's size  
more than 1's case.

Reply:
@bxf12315 please add at least this argument as a comment to the <=1 check, i.e. what and why we do ignore.

Reply:
@cristianonicolai If columnSortList size >1, it will throw exception
like this:
ErraiConsoleLogHandler.java:91 Error: IndexOutOfBoundsException: Row index: 0, Row size: 1
    at M_k_g$.pFb_g$ [as createError_0_g$] (Throwable.java:120)
    at M_k_g$.zFb_g$ [as initializeBackingError_0_g$] (Throwable.java:112)
    at M_k_g$.iFb_g$ (Throwable.java:66)
    at M_k_g$.OFb_g$ (Exception.java:29)
    at M_k_g$.Xmc_g$ (RuntimeException.java:29)
    at new M_k_g$ (IndexOutOfBoundsException.java:29)
    at N$mb_g$.Rdi_g$ [as checkRowBounds_0_g$] (AbstractHasData.java:1024)
    at N$mb_g$.Tfi_g$ [as getSubRowElement_0_g$] (AbstractCellTable.java:2285)
    at N$mb_g$.Nfi_g$ [as getKeyboardSelectedTableCellElement_0_g$] (AbstractCellTable.java:2514)
    at N$mb_g$.hgi_g$ [as onBlur_1_g$] (AbstractCellTable.java:1767)
    at N$mb_g$.tei_g$ [as onBrowserEvent_0_g$] (AbstractHasData.java:763)
    at GAi_g$ (DOM.java:1481)
    at FAi_g$ (DOM.java:1420)
    at Fki_g$ (CellBasedWidgetImplStandard.java:67)
    at HTMLDivElement.<anonymous> (CellBasedWidgetImplStandard.java:121)
    at jMf_g$ (Impl.java:309)
    at mMf_g$ (Impl.java:361)
    at HTMLDivElement.<anonymous> (Impl.java:78)
    at Nig_g$ (Node.java:291)
    at Oig_g$ (Node.java:309)
    at Fhi_g$.Hhi_g$ [as detachSectionElement_0_g$] (AbstractCellTable.java:613)
    at Fhi_g$.Jhi_g$ [as replaceAllRows_0_g$] (AbstractCellTable.java:522)
    at N$mb_g$.rfi_g$ [as createHeaders_0_g$] (AbstractCellTable.java:2408)
    at Xgi_g$.Ygi_g$ [as onModification_0_g$] (AbstractCellTable.java:855)
    at Soi_g$.Voi_g$ [as fireDelegate_0_g$] (ColumnSortList.java:317)
    at Soi_g$.Zoi_g$ [as insert_2_g$] (ColumnSortList.java:233)
    at Soi_g$._oi_g$ [as push_38_g$] (ColumnSortList.java:267)
    at EW0_g$.we__g$ [as reloadColumnSortList_0_g$] (AbstractMultiGridView.java:440)
    at xT0_g$.oc__g$ [as getData_8_g$] (AbstractMultiGridPresenter.java:189)
    at sb__g$.tb__g$ [as onRangeChanged_0_g$] (AbstractListPresenter.java:54)
    at jTj_g$.kTj_g$ [as onRangeChange_0_g$] (AbstractDataProvider.java:94)
    at MWj_g$.OWj_g$ [as dispatch_98_g$] (RangeChangeEvent.java:102)
    at MWj_g$.NWj_g$ [as dispatch_2_g$] (RangeChangeEvent.java:101)
    at MWj_g$.Ju_g$ [as dispatch_1_g$] (GwtEvent.java:76)
    at crh_g$ (EventBus.java:40)
    at Erh_g$.nrh_g$ [as doFire_0_g$] (SimpleEventBus.java:193)
    at Erh_g$.trh_g$ [as fireEvent_4_g$] (SimpleEventBus.java:88)
    at Tqh_g$.Wqh_g$ [as fireEvent_1_g$] (HandlerManager.java:127)
    at N$mb_g$.crc_g$ [as fireEvent_1_g$] (Widget.java:129)
    at PWj_g$ (RangeChangeEvent.java:54)
    at _ri_g$.Ssi_g$ [as setVisibleRange_2_g$] (HasDataPresenter.java:1488)
    at _ri_g$.Tsi_g$ [as setVisibleRangeAndClearData_0_g$] (HasDataPresenter.java:868)
    at N$mb_g$.Yei_g$ [as setVisibleRangeAndClearData_0_g$] (AbstractHasData.java:1001)
    at soi_g$.toi_g$ [as onColumnSort_0_g$] (ColumnSortEvent.java:63)
    at hoi_g$.joi_g$ [as dispatch_89_g$] (ColumnSortEvent.java:225)
    at hoi_g$.ioi_g$ [as dispatch_2_g$] (ColumnSortEvent.java:224)
    at hoi_g$.Ju_g$ [as dispatch_1_g$] (GwtEvent.java:76)
    at crh_g$ (EventBus.java:40)
    at Erh_g$.nrh_g$ [as doFire_0_g$] (SimpleEventBus.java:193)
    at Erh_g$.trh_g$ [as fireEvent_4_g$] (SimpleEventBus.java:88)
    at Tqh_g$.Wqh_g$ [as fireEvent_1_g$] (HandlerManager.java:127)
    at N$mb_g$.crc_g$ [as fireEvent_1_g$] (Widget.java:129)
    at koi_g$ (ColumnSortEvent.java:160)
    at N$mb_g$.igi_g$ [as onBrowserEvent2_0_g$] (AbstractCellTable.java:1927)
    at N$mb_g$.tei_g$ [as onBrowserEvent_0_g$] (AbstractHasData.java:775)
    at GAi_g$ (DOM.java:1481)
    at FAi_g$ (DOM.java:1420)
    at HTMLDivElement.XGi_g$ (DOMImplStandard.java:317)
    at jMf_g$ (Impl.java:309)
    at mMf_g$ (Impl.java:361)
    at HTMLDivElement.<anonymous> (Impl.java:78)

So I ignore it.